scipy/signal/tests/test_spectral.py

```diff
@@ -815,9 +838,30 @@ def test_average_all_segments(self):
         noverlap = 2
 
         f, _, P = spectrogram(x, fs, window, nperseg, noverlap)
+        _, _, H = spectrogram(x, fs, window, nperseg, noverlap, mode='complex')
         fw, Pw = welch(x, fs, window, nperseg, noverlap)
         assert_allclose(f, fw)
         assert_allclose(np.mean(P, axis=-1), Pw)
+        assert_allclose(np.mean(np.abs(H)**2, axis=-1), Pw)
+
+    def test_cross_spectrogram(self):
+        x = np.random.randn(1024)
+        y = np.random.randn(1024)
+
+        fs = 1.0
+        window = ('tukey', 0.25)
+        nperseg = 16
+        noverlap = 2
+
+        f, _, Hxx = spectrogram(x, fs, window, nperseg, noverlap,
+                                mode='complex')
+        f, _, Hyy = spectrogram(y, fs, window, nperseg, noverlap,
+                                mode='complex')
+
+        Hxy = Hxx.conj() * Hyy
+        fw, Cxy = csd(x, y, fs, window, nperseg, noverlap)
+        assert_allclose(f, fw)
+        assert_allclose(np.mean(Hxy, axis=-1), Cxy)
```

Comment:
Good call. I'll leave `atol=0` on the frequency vector checks though, since those should totally match up regardless.